In [2]:
import numpy as np
np.set_printoptions(threshold=10000,suppress=True)
import pandas as pd
import warnings
import matplotlib.pyplot as plt
warnings.filterwarnings('ignore')

from sklearn.decomposition import PCA

In [3]:
data = pd.read_csv('./data/villes.csv', sep=';')
X = data.iloc[:, 1:13].values
labels = data.iloc[:, 0].values

Question 2 : a)

In [7]:
pca = PCA(n_components=12)
pca.fit(X)
print(pca.explained_variance_ratio_)

[0.87042713 0.12064247 0.00378357 0.00254544 0.00096575 0.00052796
 0.00038435 0.00032959 0.00017116 0.00012295 0.00007647 0.00002315]


Pour conserver un minimum de 90% de l'information représentée dans le nuae initial, nous n'avons qu'à conserver les deux premiers axes. (0.87 + 0.12 = 0.99 > 0.90)

Question 2 : b)

Le premier axe correspondrait à la température des villes (les villes du sud auront des valeurs plus grandes que les villes du nord) et le deuxième axe correspondrait à stabilité de la température (les villes cotières auront des valeurs plus élevées car leur température varie peu entre l'été et l'hiver).

Question 2 : c)